wasm/jit/jit_amd64.go

```diff
@@ -766,7 +766,128 @@ func (c *amd64Compiler) compileSub(o *wazeroir.OperationSub) error {
 		return err
 	}
 
-	// x1 += x2.
+	// x1 -= x2.
+	prog := c.newProg()
+	prog.From.Type = obj.TYPE_REG
+	prog.From.Reg = x2.register
+	prog.To.Type = obj.TYPE_REG
+	prog.To.Reg = x1.register
+	prog.As = instruction
+	c.addInstruction(prog)
+
+	// We no longer need x2 register after ADD operation here,
+	// so we release it.
+	c.locationStack.releaseRegister(x2)
+	return nil
+}
+
+func (c *amd64Compiler) compileMul(o *wazeroir.OperationMul) (err error) {
+	switch o.Type {
+	case wazeroir.UnsignedTypeI32:
+		err = c.compileMulForInts(true, x86.AMULQ)
+	case wazeroir.UnsignedTypeI64:
+		err = c.compileMulForInts(false, x86.AMULQ)
+	case wazeroir.UnsignedTypeF32:
+		err = c.compileMulForFloats(x86.AMULSS)
+	case wazeroir.UnsignedTypeF64:
+		err = c.compileMulForFloats(x86.AMULSD)
+	}
+	return
+}
+
+func (c *amd64Compiler) compileMulForInts(is32Bit bool, mulInstruction obj.As) error {
+	// See https://www.felixcloutier.com/x86/mul if unfamiliar with the convention for
+	// integer multiplication. In summary, the destination operand must be on RAX
+	// register and the overflow info is stored in RDX. So, we have to ensure that
+	// 1) Previously located value on RDX must be saved to memory stack.
```

Comment:
I'll make comments like this to keep things consistent. We can later think if we want to make our own constants instead that map to these, in which case we can use RDX instead of REG_DX when abbreviating

```suggestion
	// 1) Previously located value on REG_DX must be saved to memory stack.
```